### DISCUSSION TITLE
Our customer occationally gets this error when sending files to sftpgo

### DISCUSSION THREAD

**Discussion by JanneHarju** [drakkan/sftpgo#769] [Q&A] [answered]
Here is log about the problem. Have you get this kind of errors before with sftpgo?
I don't know about used client or any other detail. Only that this is not consistent and most files is transferred as they should.

* 2022-03-21 14:40:06.006 (ESkipFile) **Upload of file 'order_020982_20220321144004_1.xml' was successful, but error occurred while setting the permissions and/or timestamp.**
* 2022-03-21 14:40:06.006 
* 2022-03-21 14:40:06.006 If the problem persists, turn off setting permissions or preserving timestamp. Alternatively you can turn on 'Ignore permission errors' option.
* 2022-03-21 14:40:06.006 The server does not support the operation.
* 2022-03-21 14:40:06.006 Error code: 8
* 2022-03-21 14:40:06.006 Error message from server: operation unsupported

**Comment by drakkan** [comment, marked as answer]
Azure blob and cloud storages in general do not support setting timestamps, if you want to support this feature you have to use the [metadata plugin](https://github.com/sftpgo/sftpgo-plugin-metadata). Your client is [WinSCP](https://winscp.net/eng/docs/message_preserve_time_perm)

**Reply by JanneHarju** [reply]
I think this was all I need to know. Thank you very much.

**Comment by JanneHarju** [comment]
I need to add one configuration what my collegue found from sftpgo. Under common-parameter group there is setstat_mode which can be set to 1 to ignore if client tries to set permission. Maybe 2 would work in our scenario where we use azure storage account. 
May I ask that if there is no plugin set for cloud based storage, then modification times are not tried to set and no errors are send to client? This is based on this "Requests for changing modification times are always executed for local/SFTP filesystems and are executed for cloud based filesystems if the target is a file and there is a metadata plugin available. A metadata plugin can be found [here](https://github.com/sftpgo/sftpgo-plugin-metadata)." from https://github.com/drakkan/sftpgo/blob/main/docs/full-configuration.md

**Reply by drakkan** [reply]
Just read the docs. If you need step-by-step assistance please consider sponsoring the project or purchasing SFTPGo from the [Azure marketplace](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/prasselsrl1645470739547.sftpgo_linux), the price is by far the lowest compared to products of the same type and is practically irrelevant to the costs of Azure infrastructure.

**Comment by JanneHarju** [comment]
We will consider it after we have solid feed back from our few projects which are starting to use this. After that I will try to convince our money guys to sponsor this product because then it is our intrest to keep this product alive.
We put this also to Azure but we use about ten different resources. It is based on Docker image hosted in Container instance. We use postgres for configure database. Then we need some network resources and that Azure storage account.
If I understand correctly your solution is based on virtual machine.
And for your information I have heard that MS is going to publish its own sftp solution to Azure in some near future.
